File: olproga/cpm_otbor/F.py
x, y = map(int, input().split())
if x == y:
    print(0)
else:
    if x < 2 * y:
        print(3)
    else:
        a, b, c = y, y, y
        ans = 0
        while a < x or b < x or c < x:
            if a <= b and a <= c:
                a = min(b + c - 1, x)
            elif b <= a and b <= c:
                b = min(a + c - 1, x)
            else:
                c = min(a + b - 1, x)
            ans += 1
        print(ans)

File: yandex_kruzhok_2025_bp/linear_algorithms/F.cpp
#include <bits/stdc++.h>

using namespace std;

using uint = unsigned int;

const uint N = 1 << 24;

vector<uint> df(N + 1, 0);
vector<uint> res(N), pref(N + 1, 0);

uint nextRand(uint a, uint b, uint &cur) {
    cur = cur * a + b;
    return cur >> 8;
}

int main() {
    uint m, q, a, b; cin >> m >> q >> a >> b;
    uint cur = 0;
    while (m--) {
        uint d = nextRand(a, b, cur);
        uint l = min(nextRand(a, b, cur), N - 1);
        uint r = min(nextRand(a, b, cur), N - 1);
        if (l > r) {swap(l, r);}
        df[l] += d;
        if (r + 1 < N) {df[r + 1] -= d;}
    }
    uint ans = 0, tmp = 0;
    for (uint i = 0; i < N; i++) {
        tmp += df[i]; res[i] = tmp;
    }
    for (uint i = 0; i < N; i++) {pref[i + 1] = pref[i] + res[i];}
    while (q--) {
        uint l = min(nextRand(a, b, cur), N - 1);
        uint r = min(nextRand(a, b, cur), N - 1);
        if (l > r) {swap(l, r);}
        ans += pref[r + 1] - pref[l];
    }
    cout << ans << endl;
    return 0;
}